core/common/src/test/scala/org/typelevel/otel4s/TextMapUpdaterSuite.scala

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2022 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s
+
+import munit.FunSuite
+
+import scala.collection.immutable
+
+class TextMapUpdaterSuite extends FunSuite {
+  // `TextMapUpdater[C]` is not implicitly summoned by this method
+  // so that it tests that instances are available in a non-generic
+  // context.
+  def check[C](tmu: TextMapUpdater[C])(initial: C)(expected: C): Unit = {
+    val res =
+      List("1" -> "one", "2" -> "two", "3" -> "three")
+        .fold(initial) { case (carrier: C, (key: String, value: String)) =>
```

Comment:
This is failing due to erasure, at least on Scala 3.2.2 and JavaScript.  Does it compile without the `: C`?

Reply:
it didn't in intelliJ, but I can test again

Reply:
this should have been `foldLeft`. I'm not even sure why `fold` compiled on whatever the default version is (2.13 I presume). maybe it inferred `Any`? :woman_shrugging: